certbot/client.py

```diff
@@ -196,13 +196,17 @@ def perform_registration(acme, config, tos_cb):
         if config.eab_kid is None or config.eab_hmac_key is None:
             raise errors.Error("Server requires external account binding. Please use --eab-kid and --eab-hmac-key.")
 
-    account_public_key = acme.client.net.key.public_key()
+        account_public_key = acme.client.net.key.public_key()
+        eab = messages.ExternalAccountBinding.from_data(account_public_key=account_public_key,
```

Comment:
This should probably be outside the "is eab required by the server" if, and inside a separate "are the necessary things available" if.

Reply:
Just pushed an updated version with two ifs. Is this what you had in mind?

Reply:
yes, very nice!

Reply:
Great, will proceed with tests and docs.